bindata/bootkube/bootstrap-manifests/etcd-member-pod.yaml

```diff
@@ -43,41 +43,38 @@ spec:
 
       source /run/etcd/environment
 
-      # TODO remove when this is handeled in setup-etcd-env
-      ETCD_IP_ADDRESS=${ETCD_IPV4_ADDRESS:-$ETCD_IPV6_ADDRESS}
-
-      [ -e /etc/ssl/etcd/system:etcd-server:${ETCD_IP_ADDRESS}.crt -a \
-        -e /etc/ssl/etcd/system:etcd-server:${ETCD_IP_ADDRESS}.key ] || \
+      [ -e /etc/ssl/etcd/system:etcd-server:${ETCD_ESCAPED_IP_ADDRESS}.crt -a \
+        -e /etc/ssl/etcd/system:etcd-server:${ETCD_ESCAPED_IP_ADDRESS}.key ] || \
         kube-client-agent \
           request \
             --kubeconfig=/etc/kubernetes/kubeconfig \
             --orgname=system:etcd-servers \
             --assetsdir=/etc/ssl/etcd \
             --dnsnames={{.EtcdServerCertDNSNames}} \
-            --commonname=system:etcd-server:${ETCD_IP_ADDRESS} \
+            --commonname=system:etcd-server:${ETCD_ESCAPED_IP_ADDRESS} \
             --ipaddrs=${ETCD_IPV4_ADDRESS},{{ .EtcdAddress.LocalHost }} \
 
-      [ -e /etc/ssl/etcd/system:etcd-peer:${ETCD_IP_ADDRESS}.crt -a \
-        -e /etc/ssl/etcd/system:etcd-peer:${ETCD_IP_ADDRESS}.key ] || \
+      [ -e /etc/ssl/etcd/system:etcd-peer:${ETCD_ESCAPED_IP_ADDRESS}.crt -a \
+        -e /etc/ssl/etcd/system:etcd-peer:${ETCD_ESCAPED_IP_ADDRESS}.key ] || \
         kube-client-agent \
           request \
             --kubeconfig=/etc/kubernetes/kubeconfig \
             --orgname=system:etcd-peers \
             --assetsdir=/etc/ssl/etcd \
             --dnsnames={{.EtcdPeerCertDNSNames}} \
-            --commonname=system:etcd-peer:${ETCD_IP_ADDRESS} \
-            --ipaddrs=${ETCD_IPV4_ADDRESS} \
+            --commonname=system:etcd-peer:${ETCD_ESCAPED_IP_ADDRESS} \
+            --ipaddrs=${ETCD_ESCAPED_IP_ADDRESS} \
 
-      [ -e /etc/ssl/etcd/system:etcd-metric:${ETCD_IP_ADDRESS}.crt -a \
-        -e /etc/ssl/etcd/system:etcd-metric:${ETCD_IP_ADDRESS}.key ] || \
+      [ -e /etc/ssl/etcd/system:etcd-metric:${ETCD_ESCAPED_IP_ADDRESS}.crt -a \
```

Comment:
I think my lack of bash knowledge is showing here.  What is the difference between this stanza and the one above?

Reply:
We are checking if the file exists. In doing so we are also validating that the values have not changed. In which case the cert would be invalid as $ETCD_ESCAPED_IP_ADDRESS is used in the IP SAN. On the master nodes, this is FQDN [1].

But logically this reads: if crt and key exist no-op (certs exist and SAN is probably still valid) or we need to request new certs.

[1] https://github.com/openshift/machine-config-operator/blob/2810704cf1e74e9a598fed158886c1fead39b33e/templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml#L71